core/src/main/java/org/elasticsearch/search/aggregations/metrics/percentiles/AbstractPercentilesParser.java

```diff
@@ -45,10 +45,11 @@ public AbstractPercentilesParser(boolean formattable) {
     }
 
     @Override
-    protected boolean token(String aggregationName, String currentFieldName, Token token, XContentParser parser,
-            ParseFieldMatcher parseFieldMatcher, Map<ParseField, Object> otherOptions) throws IOException {
+    protected boolean token(String aggregationName, String currentFieldName, Token token,
+                            ParserContext context, Map<ParseField, Object> otherOptions) throws IOException {
+        XContentParser parser = context.getParser();
         if (token == XContentParser.Token.START_ARRAY) {
-            if (parseFieldMatcher.match(currentFieldName, keysField())) {
+            if (context.getParseFieldMatcher().match(currentFieldName, keysField())) {
```

Comment:
I wonder if we should have a method in ParserContext#matchField(...) that calls `getParseFieldMatcher().match` for us?